lisa/tests/test_env_requirement.py

```diff
@@ -0,0 +1,157 @@
+from lisa import schema
+from lisa.search_space import IntRange
+from lisa.tests.test_search_space import SearchSpaceTestCase
+from lisa.testsuite import (
+    DEFAULT_REQUIREMENT,
+    TestCaseRequirement,
+    TestCaseSchema,
+    simple_requirement,
+)
+from lisa.util import constants
+
+
+class RequirementTestCase(SearchSpaceTestCase):
+    def test_supported_simple_requirement(self) -> None:
+        n6 = schema.NodeSpace.schema().load(  # type:ignore
+            {"type": constants.ENVIRONMENTS_NODES_REQUIREMENT, "coreCount": 6}
+        )
+        n6 = n6.get_min_value(n6)
+        n26 = schema.NodeSpace.schema().load(  # type:ignore
+            {
+                "type": constants.ENVIRONMENTS_NODES_REQUIREMENT,
+                "nodeCount": 2,
+                "coreCount": 6,
+            }
+        )
+        n26 = n26.get_min_value(n26)
+        n6g2 = schema.NodeSpace.schema().load(  # type:ignore
+            {
+                "type": constants.ENVIRONMENTS_NODES_REQUIREMENT,
+                "coreCount": 6,
+                "gpuCount": 2,
+            }
+        )
+        n6g2 = n6g2.get_min_value(n6g2)
+        n6g1 = schema.NodeSpace.schema().load(  # type:ignore
+            {
+                "type": constants.ENVIRONMENTS_NODES_REQUIREMENT,
+                "coreCount": 6,
+                "gpuCount": 1,
+            }
+        )
+        n6g1 = n6g1.get_min_value(n6g1)
+        n10 = schema.NodeSpace.schema().load(  # type:ignore
+            {"type": constants.ENVIRONMENTS_NODES_REQUIREMENT, "coreCount": 10}
+        )
+        n10 = n10.get_min_value(n10)
+
+        s16 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s16.environment.requirements = [n6]
+        s16g2 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s16g2.environment.requirements = [n6g2]
+        s16g1 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s16g1.environment.requirements = [n6g1]
+        s110 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s110.environment.requirements = [n10]
+        s2i6 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s2i6.environment.requirements = [n26]
+        s266 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s266.environment.requirements = [n6, n6]
+        s2610 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s2610.environment.requirements = [n6, n10]
+        s2106 = TestCaseSchema(environment=schema.Environment(), platform_type=None)
+        s2106.environment.requirements = [n10, n6]
+
+        self._verify_matrix(
+            expected_meet=[
```

Comment:
This is...not super readable. I have no idea without manually counting which boolean corresponds to which case above.

Reply:
Yes, I wrote it like data driven style, as there are too many combinations... Have no better idea.